File: market/src/main/java/wanted/market/domain/product/service/dto/response/ProductDetailResponse.java
package wanted.market.domain.product.service.dto.response;

import lombok.Builder;
import lombok.Getter;
import wanted.market.domain.product.repository.entity.Product;
import wanted.market.domain.product.repository.entity.ReservationStatus;

import java.time.LocalDateTime;

@Getter
public class ProductDetailResponse {
    private String name;

    private String email;

    private int price;

    private ReservationStatus reservationStatus;

    private int quantity;

    private String content;

    private LocalDateTime createdDateTime;

    @Builder
    private ProductDetailResponse(String name, String email, int price, ReservationStatus reservationStatus, int quantity, String content, LocalDateTime createdDateTime) {
        this.name = name;
        this.email = email;
        this.price = price;
        this.reservationStatus = reservationStatus;
        this.quantity = quantity;
        this.content = content;
        this.createdDateTime = createdDateTime;
    }

    public static ProductDetailResponse of(Product product) {
        return ProductDetailResponse.builder()
                .name(product.getName())
                .email(product.getMember().getEmail())
                .price(product.getPrice())
                .reservationStatus(product.getReservationStatus())
                .createdDateTime(product.getCreatedDateTime())
                .content(product.getContent())
                .build();
    }
}

File: src/main/kotlin/wanted/market/product/dto/ProductDetailResponse.kt
package wanted.market.product.dto

import wanted.market.member.entity.Member
import wanted.market.order.entity.Order
import wanted.market.product.entity.Product

data class ProductDetailResponse(val name: String,
                           val price: Int,
                           val productStatus: String,
                           val orderHistory: OrderHistory
){
    companion object {
        fun from(product: Product, order: Order?, seller: Member): ProductDetailResponse {
            if (order != null) {
                return ProductDetailResponse(
                    product.priceName,
                    product.price,
                    product.productStatus.value,
                    OrderHistory.from(order, seller))
            }
            return ProductDetailResponse(
                product.priceName,
                product.price,
                product.productStatus.value,
                OrderHistory())

        }
    }
}

